Naive greedy alg/scheduling/Scheduler.py

```diff
@@ -123,33 +114,47 @@ def update_schedule(self, time):
             self.req_queue.pop()
             av_servers = [server for server in av_servers if server not in job_servers]
 
-        # Add config value to only consider jobs with data < conf_value
-        jobs_by_mass = sorted(
-            self.active_jobs, key=methodcaller("remaining_mass", time)
-        )
-        while jobs_by_mass and av_servers:
-            job = jobs_by_mass[0]
-            if self._is_job_reconfigurable(job, av_servers, time):
-                av_servers = self._reconfigure_job(job, av_servers, time)
-            jobs_by_mass.pop(0)
-
-        for server in list(av_servers):
-            if not self._shutdown_server(av_servers):
-                break
-            if (
-                0.5
-                < ((len(av_servers) / len(self.servers)) ** self.conf.shutdown_weight)
-                * self.conf.shutdown_prob
-            ):
-                if random() < self.conf.shutdown_time_prob:
-                    shutdown_duration = self.conf.shutdown_time_1
+        # Reconfiguration
+        if self.reconfig_enable:
+            jobs_by_mass = sorted(
+                self.active_jobs, key=methodcaller("remaining_mass", time)
+            )
+            while jobs_by_mass and av_servers:
+                job = jobs_by_mass[0]
+                if self._is_job_reconfigurable(job, av_servers, time):
+                    av_servers = self._reconfigure_job(job, av_servers, time)
+                jobs_by_mass.pop(0)
+
+        # Turn off servers
+        if self.power_off_enable:
+            for server in list(av_servers):
+                if self.param_enable:
+                    if not self._shutdown_server(av_servers):
```

Comment:
This should be moved up above if self.param_enable, it might be the cause of the infinite loop.